### PR TITLE
Export improvements + documentation

### LINKED ISSUE #1726: [securedrop-export] Change umask on export device files to permit access from other computers with other uids
This issue should replace this one opened in the wrong repository : https://github.com/freedomofpress/securedrop-workstation/issues/822

The files exported on the usb drive are owned by uid 1000. The umask is set to 077 which makes the file readable only by uid 1000 on the system it is going to.  

https://github.com/freedomofpress/securedrop-export/blob/1d50f383f90510fb2f545659f39f2d03085b805f/securedrop_export/export.py#L79

In our case, we have Journalists working on Debian workstations, with some central LDAP authentication. A journalist trying to read the export files from securedrop on his daily workstation cannot get access to the files.

I do not see a lot of scenarios where the file permissions on the export stick improves security in any way. 
The usb drive is encrypted, so the user first needs to unlock it.
The daily computer should not be trusted. uid 0 would be able to read the files once they are mounted on the daily computer and don't care about the file system permission.

The only situations where the file permissions could increase security would be accessing the file from some unprivileged user on the system, but the exported files are already meant to become public, so I would evaluate the risk as acceptable in my situation.

On top of that, the USB drive is mounted by the system under /media/$USERNAME/ with no permissions for group or others (tested on debian bullseye but I hope it's standart). So the permissions on the filesystem on top of it don't really matter for this attack vector. An unprivileged user would not be able to read the file, if another uid mounted the usb disk.

So I would propose to change the `os.umask(0o077)` into `os.umask(0o022)` to give permissions to others to read the files, so the journalists can at least copy files to their working environment independently from their uid, when the device got mounted by themselves.

### PR DESCRIPTION
**Pull request by rocodes**
## Status

Ready for review
- Blocked by #1873

## Description

Some stragglers:

- Remove `encryption_method` from archive.py (looks like a regression maybe based on rebasing/squashing - was meant to be fully removed during VeraCrypt work)
- Fix UI issue where `EXPORT_ERROR` did not update the UI (Seen once in testing by cfm:  https://github.com/freedomofpress/securedrop-client/pull/1777#pullrequestreview-1894631155) 
- Update export README to include veracrypt instructions and note about compatible printers

And one new change:
- Adjust umask only before writing to the export USB (Fixes #1726) - Per [sec eng approval ](https://github.com/freedomofpress/securedrop-client/issues/1726#issuecomment-1932771841) and discussion with previous sec eng. 

## Test Plan
- [x] CI passes
- [x] Visual review
- [x] Rapid testing: export to USB works on LUKS and VeraCrypt
- [x] Files on export USB have permission `644` (#1726)
- [x] Export still succeeds on drives unlocked with `sudo`

## Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:

 - [ ] I have tested these changes in the appropriate Qubes environment
 - [ ] I do not have an appropriate Qubes OS workstation set up (the reviewer will need to test these changes)
 - [ ] These changes should not need testing in Qubes

If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:

 - [ ] I have updated the [AppArmor profile](https://github.com/freedomofpress/securedrop-client/blob/HEAD/files/usr.bin.securedrop-client)
 - [ ] No update to the AppArmor profile is required for these changes
 - [ ] I don't know and would appreciate guidance

If these changes modify the database schema, you should include a database migration. Please check as applicable:

 - [ ] I have written a migration and upgraded a test database based on `main` and confirmed that the migration is [self-contained] and applies cleanly
 - [ ] I have written a migration but have not upgraded a test database based on `main` and would like the reviewer to do so
 - [ ] I need help writing a database migration
 - [ ] No database schema changes are needed

[self-contained]: https://github.com/freedomofpress/securedrop-client#generating-and-running-database-migrations

### CODE REVIEW

**Comment by rocodes** [comment]
(Will need rebase when #1873 lands)

File: client/securedrop_client/gui/conversation/export/export_wizard_page.py
Comment (rocodes) [inline client/securedrop_client/gui/conversation/export/export_wizard_page.py:408]: This will address the "Hanging" at "Working...." issue that cfm saw during ERROR_EXPORT. (Addresses the following very specific case: Unlocked, attached device -> launch export wizard -> ERROR_EXPORT)

**Comment by cfm** [comment]
Good catches; this looks good.  But I've made explicit that testing is blocked on #1873.

**Comment by rocodes** [comment]
I've gone through a quick test plan myself:
- [x] LUKS and VeraCrypt export are successful
- [x] LUKS and VeraCrypt successful when drive is unlocked with `sudo` (but see notes on mounting below)
- [x] Files on the USB have 644 permissions 
- [x] When a USB is unlocked, then the export wizard is triggered, then there's an error (`ERROR_EXPORT`), the screen transitions from "Working..." to "Export Failed" (resolve issue observed once during testing)  
- **Note** Export fails if the user manually mounts the drive as root at the commandline.  The user sees the message "Export failed." There's no need for users to do any mounting at all even with VC drives, so this is just trying to catch an edge case; I suggest either documenting this in our docs, or if that doesn't seem good enough, I can add logic to detect if the reason the export failed is a permissions error) and add a more specific error message.

File: export/README.md
Comment (cfm) [inline export/README.md:25]: Suggested revision for clarity:

```suggestion
Partition a drive (either the MBR/DOS or the GPT partition scheme is fine)
and encrypt exactly one partition with LUKS.  On Linux, you can
[follow these instructions](https://docs.securedrop.org/en/stable/set_up_transfer_and_export_device.html#create-usb-transfer-device)
to use GNOME Disks to set up your drive.  You can also use
[cryptsetup](https://linux.die.net/man/8/cryptsetup):
```

File: export/README.md
Comment (cfm) [inline export/README.md:28]: Nit: Can we standardize on either `lsblk -p` or `sudo fdisk -l` to keep the lookup consistent between methods?

**Review by cfm — CHANGES_REQUESTED** [review]
Test plan checks out.  @rocodes, I've left some documentation suggestions inline, but I'll be happy to approve this once you've decided whether to keep or discard them.

File: export/README.md
Comment (rocodes) [reply export/README.md]: updated word for word in https://github.com/freedomofpress/securedrop-client/commit/1b44601ed41c1679d7241bbe5e6758ae6f4a8687 :)

**Review by cfm — APPROVED** [review]
Thanks, @rocodes!

**Pull request merged**
